ang/civicase/CaseDetailsPeople.js

```diff
@@ -23,113 +43,85 @@
-      }
-      var caseRoles = $scope.rolesAlphaFilter ? [] : _.cloneDeep($scope.allRoles),
-        allRoles = _.cloneDeep($scope.allRoles),
-        selected = getSelectedContacts('roles', true);
-      if ($scope.rolesFilter) {
-        caseRoles = $scope.rolesFilter === 'client' ? [] : [_.findWhere(caseRoles, {name: $scope.rolesFilter})];
-      }
-      var relDesc = [];
-      // get relationship descriptions
-      if(item["api.Relationship.get"]) {
-        _.each(item["api.Relationship.get"].values, function (relationship) {
-          relDesc[relationship.contact_id_b + '_' + relationship.relationship_type_id] = relationship['description'] ? relationship['description'] : '';
-        });
-      }
-      _.each(item.contacts, function (contact) {
-        var role = contact.relationship_type_id ? _.findWhere(caseRoles, {relationship_type_id: contact.relationship_type_id}) : null;
-        if ((!role || role.contact_id) && contact.relationship_type_id) {
-          role = _.cloneDeep(_.findWhere(allRoles, {relationship_type_id: contact.relationship_type_id}));
-          if (!$scope.rolesFilter || role.name === $scope.rolesFilter) {
-            caseRoles.push(role);
-          }
-        }
-        // Apply filters
-        if ($scope.rolesAlphaFilter && contact.display_name.toUpperCase().indexOf($scope.rolesAlphaFilter) < 0) {
-          if (role) _.pull(caseRoles, role);
-        } else if (role) {
-          if (!$scope.rolesFilter || role.name === $scope.rolesFilter) {
-            $.extend(role, {checked: selected.indexOf(contact.contact_id) >= 0}, contact);
-          }
-        } else if (!$scope.rolesFilter || $scope.rolesFilter == 'client') {
-          caseRoles.push($.extend({role: ts('Client'), checked: selected.indexOf(contact.contact_id) >= 0}, contact));
-        }
-      });
-      _.each(caseRoles, function (role, index) {
-        if(role && role.role != 'Client' && (role.contact_id + '_' + role.relationship_type_id in relDesc)) {
-          caseRoles[index]['desc'] = relDesc[role.contact_id + '_' + role.relationship_type_id];
-        }
-      });
-      $scope.rolesCount = caseRoles.length;
-      // Apply pager
-      if ($scope.rolesCount <= (25 * ($scope.rolesPage - 1))) {
-        // Reset if out of range
-        $scope.rolesPage = 1;
-      }
-      $scope.roles = _.slice(caseRoles, (25 * ($scope.rolesPage - 1)), 25 * $scope.rolesPage);
+    $scope.setSelectionMode = function (mode, tab) {
+      $scope[tab + 'SelectionMode'] = mode;
     };
 
-    $scope.setSelectionMode = function(mode, tab) {
-      $scope[tab + 'SelectionMode'] = mode;
+    $scope.setTab = function (tab) {
+      $scope.tab = tab;
+    };
+
+    $scope.setLetterFilter = function (letter, tab) {
+      if ($scope[tab + 'AlphaFilter'] === letter) {
+        $scope[tab + 'AlphaFilter'] = '';
+      } else {
+        $scope[tab + 'AlphaFilter'] = letter;
+      }
+      if (tab === 'roles') {
+        $scope.getCaseRoles();
```

Comment:
If we go with the `$scope.tabs = {}` you can have something like this instead:

```js
$scope.tabs = {
  roles: { get: getCaseRoles },
  relations: { get: getRelations },
};

$scope.setLetterFilter = function (letter, tab) {
  // ...

  $scope.tabs[tab].get();
};
```

Also `get`, `getRelations`, and `getCaseRoles` are not the proper names for these functions. When a function starts with `get` there's an expectation that it will return a value, and this is not the case. It would be good to rename them to `loadRelations`, etc. but only if you have time and think this is feasible.

Reply:
As per plan. We won't be fixing (refactoring) the logic of the already present functions if they work fine. As this will take too much time and disturb the sprint plan. However it will be a good add on to have documentation for these function.